node/src/reactor/main_reactor/keep_up.rs

```
@@ -387,26 +349,30 @@ impl MainReactor {
                     Err(GetEraValidatorsError::RootNotFound),
                 ) => Err(vec![
                     (
-                        parent_metadata.parent_hash,
-                        parent_metadata.parent_state_hash,
+                        global_states_metadata.before_hash,
+                        global_states_metadata.before_state_hash,
+                    ),
+                    (
+                        global_states_metadata.after_hash,
+                        global_states_metadata.after_state_hash,
                     ),
-                    (block_hash, parent_metadata.global_state_hash),
                 ]),
-                // The block's global state was missing - return the hashes.
-                (Ok(_), Err(GetEraValidatorsError::RootNotFound)) => {
-                    Err(vec![(block_hash, parent_metadata.global_state_hash)])
-                }
-                // The parent's global state was missing - return the hashes.
+                // The after-block's global state was missing - return the hashes.
+                (Ok(_), Err(GetEraValidatorsError::RootNotFound)) => Err(vec![(
+                    global_states_metadata.after_hash,
+                    global_states_metadata.after_state_hash,
+                )]),
+                // The before-block's global state was missing - return the hashes.
                 (Err(GetEraValidatorsError::RootNotFound), Ok(_)) => Err(vec![(
-                    parent_metadata.parent_hash,
-                    parent_metadata.parent_state_hash,
+                    global_states_metadata.before_hash,
+                    global_states_metadata.before_state_hash,
                 )]),
                 // We got some error other than `RootNotFound` - just log the error and don't
                 // synchronize anything.
-                (parent_result, block_result) => {
+                (before_result, after_result) => {
```

nit: I guess if we keep getting a different error other than `RootNotFound` the node would be stuck in a loop requesting the same sync leap over and over again. It's not a big deal since we would see the error in the log but I'm wondering if we want to terminate the node in this this happens repeatedly?

---

I think this is fine as is. Looking at the other error variants, they would point to global state and/or engine state being corrupted, in which case restarts wouldn't help.

---

But it's a good point that we might want to bring this to the operator's attention ASAP, and just logging the error without doing anything else might not be the best strategy :thinking: 
I guess we might want to either propagate the error somewhere (where, though?) or just `fatal!` outright.